src/test/scala/is/hail/methods/AggregatorSuite.scala

```diff
@@ -123,6 +126,31 @@ class AggregatorSuite extends SparkSuite {
     p.check()
   }
 
+  @Test def testMaxMin() {
+    val rdd1 = sc.parallelize(Seq(
+      Annotation("a",  0, null, -1, -1.0, null),
+      Annotation("a", -1, null, -2, 1.0, null)), numSlices = 1)
+
+    val rdd2 = sc.parallelize(Seq(
+      Annotation("a",  1, -1, -3, null, null),
+      Annotation("a",  0, -2, -4, 0.0, null)), numSlices = 1)
+
+    val rdd = rdd1.union(rdd2)
```

Comment:
Why do you need two per partition?  You just need to test each side being null.  Why not:

null, null, 1
1, null, null

parallelize splits evenly:

```
scala> sc.parallelize(Seq(1, 2, 3, 4), 2).mapPartitions { it => Iterator(it.length) }.collect()
res1: Array[Int] = Array(2, 2)
```